backend/app/store/image/image.go

```diff
@@ -140,7 +140,10 @@ func (s *Service) Submit(idsFn func() []string) {
 }
 
 // ExtractPictures gets list of images from the doc html and convert from urls to ids, i.e. user/pic.png
-func (s *Service) ExtractPictures(commentHTML string) (ids []string) {
+// Proxied images are returned only if the flag is set: this is used in image check on post preview and load,
+// as proxied images have lazy loading and wouldn't be present on disk but still valid as they will be loaded
+// the first time someone requests them.
+func (s *Service) ExtractPictures(commentHTML string, returnProxied bool) (ids []string) {
```

Comment:
this new bool is used in multiple places and each place needs a comment about what this true/false means. How about making a small "enum" type with "image.DIRECT" and "image.PROXIED" values (consts)? Should make the api more readable

Reply:
@umputun please take a look at the current version.